arcgis-ios-sdk-samples/Edit data/Edit and sync features/EditAndSyncFeaturesViewController.swift

```diff
@@ -0,0 +1,277 @@
+               
+                // Request a job to generate the geodatabase.
+                let generateGeodatabaseJob = self.geodatabaseSyncTask.generateJob(with: params, downloadFileURL: downloadFileURL)
+                self.generateJob = generateGeodatabaseJob
+                generateGeodatabaseJob.start(
+                    statusHandler: { (status: AGSJobStatus) in
+                        SVProgressHUD.show(withStatus: status.statusString()) //Show job status.
+                    },
+                    completion: { [weak self] (_, error: Error?) in
+                        SVProgressHUD.dismiss()
+                        
+                        if let error = error {
+                            self?.presentAlert(error: error)
+                        } else {
+                            // Load the geodatabase when the job is done.
+                            self?.geodatabase = generateGeodatabaseJob.result
+                            self?.geodatabase.load { [weak self] (error: Error?) in
+                                self?.geodatabaseDidLoad()
+                            }
+                        }
+                    }
+                )
+            } else {
+                self!.presentAlert(title: "Could not generate default parameters: \(error!)")
+            }
+        }
+    }
+    
+    func syncGeodatabase() {
+        clearSelection()
+        barButtonItem.isEnabled = false
+        selectedFeature = nil
+        
+        // Create parameters for the sync task.
+        let syncGeodatabaseParameters = AGSSyncGeodatabaseParameters()
+        syncGeodatabaseParameters.geodatabaseSyncDirection = .bidirectional
+        syncGeodatabaseParameters.rollbackOnFailure = false
+        
+        // Specify the layer IDs of the feature tables to sync.
+        for geodatabaseFeatureTable in geodatabase.geodatabaseFeatureTables {
+            let serviceLayerId = geodatabaseFeatureTable.serviceLayerID
+            let syncLayerOption = AGSSyncLayerOption(layerID: serviceLayerId, syncDirection: .bidirectional)
+            syncGeodatabaseParameters.layerOptions.append(syncLayerOption)
+        }
+        
+        // Create a sync job with the parameters and start it.
+        let syncGeodatabaseJob = geodatabaseSyncTask.syncJob(with: syncGeodatabaseParameters, geodatabase: geodatabase)
+        self.syncJob = syncGeodatabaseJob
+        syncGeodatabaseJob.start(statusHandler: { (status: AGSJobStatus) in
+            SVProgressHUD.show(withStatus: status.statusString())
+        }, completion: { [weak self] (result: [AGSSyncLayerResult]?, error: Error?) in
+            SVProgressHUD.dismiss()
+            if let error = error {
+                self?.presentAlert(error: error)
+            } else {
+                self?.geodatabaseDidSync()
+            }
+        })
+    }
+    
+    @IBAction func generateOrSync() {
+        if barButtonItem.title == "Sync geodatabase" {
```

Comment:
Using the string to determine state is somewhat fragile.

Ideally, one might define the state of the app with either a boolean or an enum, and just reflect the state with a string rather than using the string itself to define the state.

However, in this case I would be Ok with it if you defined the "Sync geodatabase" string as a `private let` file level or class level let, and then replace the instances of "Sync geodatabase" with a reference to that.

Reply:
I don't think I'll be able to define it as a `private let` at file or class level because the title is set to "Generate geodatabase" in the storyboard. I believe setting it to "Sync geodatabase" at file/class level as a `private let` would not allow me to change the title when needed.

Reply:
Hi @vquach2404: See https://github.com/Esri/arcgis-runtime-samples-ios/commit/6484386c4ed54d9caf01b0a8a011503b6bc1b322#diff-857df2a38391d55c8b3258543acb5541 to see what I mean.

If this is also set in the Storyboard, then this should really be redone as a boolean state variable, or an enum (though the enum would only have 2 options).